### PR TITLE
Calibration Utility Updates

### PR DESCRIPTION
**Pull request by landmanbester**
- [ ] Tests added / passed

  ```bash
  $ py.test -v -s africanus
  ```

  If the pep8 tests fail, the quickest way to correct
  this is to run `autopep8` and then `flake8` and
  `pycodestyle` to fix the remaining issues.

  ```
  $ pip install -U autopep8 flake8 pycodestyle
  $ autopep8 -r -i africanus
  $ flake8 africanus
  $ pycodestyle africanus
  ```

- [ ] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

Add a check to make sure the correlation axis is never larger than 2

### CODE REVIEW

File: africanus/calibration/utils/utils.py [hunk 1]
```diff
@@ -39,15 +39,23 @@ def check_type(jones, vis, vis_type='vis'):
...
     time_bin_indices = np.zeros(n_time, dtype=np.int32)
     time_bin_indices[1::] = np.cumsum(time_bin_counts)[0:-1]
     time_bin_indices = time_bin_indices.astype(np.int32)
     time_bin_counts = time_bin_counts.astype(np.int32)
+    if daskify_idx:
+        import dask.array as da
+        time_bin_indices = da.from_array(
+            time_bin_indices, chunks=utimes_per_chunk)
+        time_bin_counts = da.from_array(
+            time_bin_counts, chunks=utimes_per_chunk)
```
Comment (sjperkins) [inline africanus/calibration/utils/utils.py:58]: Hehe, I'm so on the fence about this. On the one hand it's probably very convenient, on the other hand it feels like this dask wrapping functionality should be elsewhere. A `dask_chunkify_rows` wrapper around `chunkify_rows` might be "better"? But I don't feel strongly about this, its a utility function after all so happy to go with whatever you decide here.

File: africanus/gps/kernels.py [hunk 2]
```diff
@@ -33,13 +33,13 @@ def exponential_squared(x, xp, sigmaf, l, pspec=False):  # noqa: E741
     if pspec:
         N, D = x.shape
         if D != 1:
-            raise NotImplementedError("Only 1D pspecs supported")
+            raise (NotImplementedError, "Only 1D pspecs supported")
```
[the remark this answers is not in the document]
Comment (landmanbester) [reply africanus/gps/kernels.py:36]: Oops, not sure how that slipped in

File: africanus/calibration/utils/utils.py (hunk 1 above)
Comment (landmanbester) [reply africanus/calibration/utils/utils.py]: I have removed it, was just getting repetitive but I agree it's better to do it externally

**Comment by sjperkins** [comment]
I forgot to ask for the update to `HISTORY.rst`. If you prefix the commit with `[skip ci]` it'll avoid another ci run.

**Comment by landmanbester** [comment]
Oops, forgot about that. Thanks @sjperkins

**Pull request merged**